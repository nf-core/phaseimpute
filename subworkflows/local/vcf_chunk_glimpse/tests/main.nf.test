nextflow_workflow {

    name "Test Subworkflow VCF_CHUNK_GLIMPSE"
    script "../main.nf"

    config "./nextflow.config"

    workflow "VCF_CHUNK_GLIMPSE"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "subworkflows/vcf_chunk_glimpse"
    tag "vcf_chunk_glimpse"

    tag "glimpse"
    tag "glimpse/chunk"
    tag "glimpse2"
    tag "glimpse2/chunk"

    test("Chunks with Map") {
        when {
            params {
                max_cpus   = 2
                max_memory = '2.GB'
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [id: "1000GP", chr: "chr22"],
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr22/1000GP.chr22.s.norel.vcf.gz", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr22/1000GP.chr22.s.norel.vcf.gz.csi", checkIfExist:true),
                    ],
                    [
                        [id: "1000GP", chr: "chr21"],
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr21/1000GP.chr21.s.norel.vcf.gz", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr21/1000GP.chr21.s.norel.vcf.gz.csi", checkIfExist:true),
                    ]
                )
                input[1] = Channel.of(
                    [
                        [chr: "chr22"],
                        file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38_22.map", checkIfExist:true)
                    ],
                    [
                        [chr: "chr21"],
                        file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38_21.map", checkIfExist:true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert snapshot(workflow.out.chunks.collect{
                    path(it[1]).readLines()
                    }).match("chunksWithMap")
                }
            )
        }
    }

    test("Chunks without Map") {
        when {
            params {
                max_cpus   = 2
                max_memory = '2.GB'
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [id: "1000GP", chr: "chr22"],
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr22/1000GP.chr22.s.norel.vcf.gz", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr22/1000GP.chr22.s.norel.vcf.gz.csi", checkIfExist:true),
                    ],
                    [
                        [id: "1000GP", chr: "chr21"],
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr21/1000GP.chr21.s.norel.vcf.gz", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/panel/chr21/1000GP.chr21.s.norel.vcf.gz.csi", checkIfExist:true),
                    ]
                )
                input[1] = Channel.of(
                    [[chr: "chr22"], []],
                    [[chr: "chr21"], []]
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert snapshot(workflow.out.chunks.collect{
                    path(it[1]).readLines()
                    }).match("chunksWithoutMap")
                }
            )
        }
    }
}
