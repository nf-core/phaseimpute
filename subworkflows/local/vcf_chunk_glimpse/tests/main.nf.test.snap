{
    "chunksWithoutMap": {
        "content": [
            [
                [
                    "0\tchr21\tchr21:16570070-16595525\tchr21:16570070-16590513\t20444\t419",
                    "1\tchr21\tchr21:16585483-16609998\tchr21:16590521-16609998\t19478\t417"
                ],
                [
                    "0\tchr22\tchr22:16570065-16597215\tchr22:16570065-16592216\t22152\t452",
                    "1\tchr22\tchr22:16587172-16609999\tchr22:16592229-16609999\t17771\t451"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-07-18T11:26:35.422657952"
    },
    "Chunks without Map": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "1000GP_chr21.txt:md5,64ca4a1655363e4a4e558836e5ac12f9"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "1000GP_chr22.txt:md5,3344e171251722cf58ae31136da223ac"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21",
                        "16570070",
                        "16595525"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21",
                        "16585483",
                        "16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22",
                        "16570065",
                        "16597215"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22",
                        "16587172",
                        "16609999"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16595525",
                        "chr21:16570070-16590513"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16585483-16609998",
                        "chr21:16590521-16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16597215",
                        "chr22:16570065-16592216"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16587172-16609999",
                        "chr22:16592229-16609999"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16609998",
                        "chr21:16570070-16590517"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16609998",
                        "chr21:16590518-16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16609999",
                        "chr22:16570065-16592222"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16609999",
                        "chr22:16592223-16609999"
                    ]
                ],
                "4": [
                    [
                        [
                            
                        ]
                    ]
                ],
                "5": [
                    "versions.yml:md5,1ae67fb38ed979f02f47d5ecb8a85f0d",
                    "versions.yml:md5,1ae67fb38ed979f02f47d5ecb8a85f0d",
                    "versions.yml:md5,7d277747b107043dd31d3aef18045eef"
                ],
                "binary": [
                    [
                        [
                            
                        ]
                    ]
                ],
                "chunks": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "1000GP_chr21.txt:md5,64ca4a1655363e4a4e558836e5ac12f9"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "1000GP_chr22.txt:md5,3344e171251722cf58ae31136da223ac"
                    ]
                ],
                "chunks_glimpse1": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16595525",
                        "chr21:16570070-16590513"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16585483-16609998",
                        "chr21:16590521-16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16597215",
                        "chr22:16570065-16592216"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16587172-16609999",
                        "chr22:16592229-16609999"
                    ]
                ],
                "chunks_glimpse2": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16609998",
                        "chr21:16570070-16590517"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16609998",
                        "chr21:16590518-16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16609999",
                        "chr22:16570065-16592222"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16609999",
                        "chr22:16592223-16609999"
                    ]
                ],
                "chunks_quilt": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21",
                        "16570070",
                        "16595525"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21",
                        "16585483",
                        "16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22",
                        "16570065",
                        "16597215"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22",
                        "16587172",
                        "16609999"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1ae67fb38ed979f02f47d5ecb8a85f0d",
                    "versions.yml:md5,1ae67fb38ed979f02f47d5ecb8a85f0d",
                    "versions.yml:md5,7d277747b107043dd31d3aef18045eef"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-07-18T11:26:34.820936365"
    },
    "Chunks with Map": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "1000GP_chr21.txt:md5,64ca4a1655363e4a4e558836e5ac12f9"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "1000GP_chr22.txt:md5,3344e171251722cf58ae31136da223ac"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21",
                        "16570070",
                        "16595525"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21",
                        "16585483",
                        "16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22",
                        "16570065",
                        "16597215"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22",
                        "16587172",
                        "16609999"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16595525",
                        "chr21:16570070-16590513"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16585483-16609998",
                        "chr21:16590521-16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16597215",
                        "chr22:16570065-16592216"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16587172-16609999",
                        "chr22:16592229-16609999"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16609998",
                        "chr21:16570070-16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16609999",
                        "chr22:16570065-16609999"
                    ]
                ],
                "4": [
                    [
                        [
                            
                        ]
                    ]
                ],
                "5": [
                    "versions.yml:md5,1ae67fb38ed979f02f47d5ecb8a85f0d",
                    "versions.yml:md5,1ae67fb38ed979f02f47d5ecb8a85f0d",
                    "versions.yml:md5,7d277747b107043dd31d3aef18045eef"
                ],
                "binary": [
                    [
                        [
                            
                        ]
                    ]
                ],
                "chunks": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "1000GP_chr21.txt:md5,64ca4a1655363e4a4e558836e5ac12f9"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "1000GP_chr22.txt:md5,3344e171251722cf58ae31136da223ac"
                    ]
                ],
                "chunks_glimpse1": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16595525",
                        "chr21:16570070-16590513"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16585483-16609998",
                        "chr21:16590521-16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16597215",
                        "chr22:16570065-16592216"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16587172-16609999",
                        "chr22:16592229-16609999"
                    ]
                ],
                "chunks_glimpse2": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21:16570070-16609998",
                        "chr21:16570070-16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22:16570065-16609999",
                        "chr22:16570065-16609999"
                    ]
                ],
                "chunks_quilt": [
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21",
                        "16570070",
                        "16595525"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr21"
                        },
                        "chr21",
                        "16585483",
                        "16609998"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22",
                        "16570065",
                        "16597215"
                    ],
                    [
                        {
                            "id": "1000GP",
                            "chr": "chr22"
                        },
                        "chr22",
                        "16587172",
                        "16609999"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1ae67fb38ed979f02f47d5ecb8a85f0d",
                    "versions.yml:md5,1ae67fb38ed979f02f47d5ecb8a85f0d",
                    "versions.yml:md5,7d277747b107043dd31d3aef18045eef"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-07-18T11:26:24.155279577"
    },
    "chunksWithMap": {
        "content": [
            [
                [
                    "0\tchr21\tchr21:16570070-16595525\tchr21:16570070-16590513\t20444\t419",
                    "1\tchr21\tchr21:16585483-16609998\tchr21:16590521-16609998\t19478\t417"
                ],
                [
                    "0\tchr22\tchr22:16570065-16597215\tchr22:16570065-16592216\t22152\t452",
                    "1\tchr22\tchr22:16587172-16609999\tchr22:16592229-16609999\t17771\t451"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-07-18T11:26:24.609554836"
    }
}